scripts/tls/README.md

```diff
@@ -0,0 +1,61 @@
+## gRPC TLS Generation
+
+The Matchbox gRPC API allows clients (`terraform-provider-matchbox`) to create and update Matchbox resources. TLS credentials are used for client authentication and to establish a secure communication channel. When the gRPC API is [enabled](../../Documentation/deployment.md#customization), the server requires a TLS server certificate, key, and CA certificate ([locations](../../Documentation/config.md#files-and-directories)).
+
+The `cert-gen` helper script generates a self-signed CA, server certificate, and client certificate. **Prefer your organizations' PKI, if possible**
+
+Navigate to the `scripts/tls` directory.
+
+```sh
+$ cd scripts/tls
+```
+
+Export `SAN` to set the Subject Alt Names which should be used in certificates. Provide the fully qualified domain name or IP (discouraged) where Matchbox will be installed.
+
+```sh
+# DNS or IP Subject Alt Names where matchbox runs
+$ export SAN=DNS.1:matchbox.example.com,IP.1:172.18.0.2
+```
+
+Generate a `ca.crt`, `server.crt`, `server.key`, `client.crt`, and `client.key`.
+
+```sh
+$ ./cert-gen
+Creating FAKE CA, server cert/key, and client cert/key...
+...
+...
+...
+******************************************************************
+WARNING: Generated TLS credentials are ONLY SUITABLE FOR EXAMPLES!
```

Comment:
I've relaxed this warning to be more allowing and just dropped the suitable only for examples text.